nncf/quantization/quantize_model.py

```diff
@@ -36,6 +36,16 @@
 TTensor = TypeVar("TTensor")
 
 
+def _update_advanced_quantization_parameters(advanced_parameters, calibration_dataset):
+    batch_size = calibration_dataset.get_batch_size()
```

Comment:
It seemed to me that we've agreed that we should not provide any options to users that regulate the behaviour based on the batch.  But it looks like the `statistics_per_sample` option regulates this behaviour latently. Why so?
@alexsu52, are there any other agreements?

Reply:
Also, I thought that we'd agreed that batch size calculation should be calculated internally while subset calculation and statistics collection. Not in pipeline creation step.